### ISSUE TITLE
Add afero.Symlink

### ISSUE THREAD

**Issue by relnod** [spf13/afero#174]
Hi,
`afero.Symlink()` does not seem to be implemented. Are there any plans to add this method?
See [os.Symlink](https://golang.org/pkg/os/#Symlink) for reference.
If you point me to the right direction, I could aso try to implement it myself.

**Comment by twpayne-da** [comment]
I also need this feature, and `os.ReadLink` too. Any plans to implement it or would you prefer a PR?

**Comment by millerlogic** [comment]
I agree, I also need ReadLink. I suppose it will need to be done like [lstat](https://godoc.org/github.com/spf13/afero#Lstater) so it doesn't break a bunch of code.

**Comment by mpictor** [comment]
Not sure why we should be limited to a real filesystem for `Symlink()`/`Readlink()`/`Lstat()`. I'll need these in my use case, which is recording all fs activity that takes place so it can be replicated for testing. This will involve a lot of symlinks...

**Comment by mpictor** [comment]
Quick (and potentially dirty) addition of Symlink/Readlink functionality: https://github.com/mpictor/afero/tree/symlinks

Suggestions/critiques/tests/etc welcome ;)

**Comment by cumminsd** [comment]
This would be awesome! I use Afero to test all my file system code and need to add some features to process SymLinks.

**Comment by mpictor** [comment]
I've added unit tests and an `EvaluateSymlinks` function. I think it's complete for unix-like OSes but probably not for windows. If someone could test on windows, that'd be great :smile:

**Comment by cumminsd** [comment]
Will it let you create symlinks too? Especially in the Memory File System?

I have JSON files that describe various scenarios in the file system. (Folders with files with various permissions, etc.) I have a tool that builds a Memory File System from those JSON files and then I run a suite of tests against those scenarios. Would LOVE to be able to create symlinks to test against!

**Comment by mpictor** [comment]
It won't create symlinks in the memory fs. I looked at memoryfs briefly but didn't understand what was going on there, so I just implemented Symlinker and Readlinker wherever Lstater already existed.

Don't think I'll actually be using memory fs, as my (current) plan is to record fs activity to a protobuf, then replay in testing. I may have to change my plan up, depending on how many TB of data are involved - this won't be very feasible for fs activity > 0.0001 TerrorByte :wink:

**Comment by mpictor** [comment]
I submitted a related PR, but Travis-ci and Appveyor both error out. Not sure when I'll find time to dig into it. :/

**Comment by mpictor** [comment]
Fixed the windows and linux tests on #212, but OSX is still failing. Any help would be appreciated!

```
    	link_test.go:215: lstat private/var: no such file or directory
```

**Comment by kargakis** [comment]
Fixed by https://github.com/spf13/afero/pull/228
